### ISSUE TITLE
Notify for type unsafe NBT with configurable severity

### ISSUE THREAD

**Issue by MulverineX** [SpyglassMC/Spyglass#1315]
Defaulting to an "info" highlight (gray dashed underline for unknown keys), with an option to increase severity to a warning or an error. 

This is specifically for mechanisms like `mcdoc:custom_item_data` to increase visibility to users that strict typing is now available.

**Comment by misode** [comment]
I disagree that it should show any diagnostic at all. It should be any by default. We can still increase the visibility of mcdoc in the future by: (1) highlighting it in the readme and (2) implementing go to definition of vanilla field names.

**Comment by MulverineX** [comment]
Alright, this can be off by default
